Sessionless Product Details (#67)
planet-nine-app/sessionless: Sessionless Product Detail.md

# Sessionless Product Details

**Introduction & Base Definitions**

> Authentication protocols (auth protocols) are gatekeepers that verify the identity of a person to give them access to specific programs or data. They include a set of rules or procedures that govern how data is transmitted across electronic devices. The most common uses of auth protocols are passwords, but others include one-time-passwords, biometrics, and magic links. 

Reviewer: Let's include a link to this page: https://www.okta.com/identity-101/identification-vs-authentication/#:~:text=Identification%20is%20the%20act%20of,access%20to%20systems%20or%20privileges.

Author: Updated with latest merge

- An Application Programming Interface (API) is  a software intermediary that allows two  applications to share data with each other
- An API Key is  a unique code that identifies an application or user

> - An API Token is a piece of code that contains information about a user and is used to secure access to an API

Reviewer: This isn't quite correct. In most cases API Key and API Token are synonymous. A JWT (JSON Web Token) does contain information about a user as you've described here

Author: Updated with latest merge

> - Encryption is the process of turning data into code to prevent unauthorized access

Reviewer: ```suggestion
- Encryption is the process of encoding data to prevent unauthorized access
```

Author: My only concern here is that some people may not understand what "encoding" means.

Reviewer: that's fair. My issue is that "turning data into code" implies code to me. e.g. 

```
if(true) {
    doSomething();
}
```

maybe just "Encryption is the process of preventing data from unauthorized access"

Reviewer 2: I agree, the `code` word might be confusing and I also see it that way.

How about:
```
Encryption is the process of converting information into a secret format to prevent unauthorized access
```

**Problem Statement**

> Many auth protocols are client-server, where the client supplies some secret information (usually name and email address) to authenticate requests.

Reviewer: ```suggestion
Many auth protocols are client-server, where the client supplies some secret information (usually name and email address with a password) to authenticate requests.
```

There are multiple problems with this set up:
- Requires users to enter in personally identifiable information which can be hacked presenting legal risks 

> - Requires encryption to prevent sensitive data leaks using https which slows and limits authentication opportunities (can’t connect to anything other than standard computers)

Reviewer: This is worded such that it sounds like the data leaks happen because of https. Maybe something like, "Encryption protocols, like https, provide some protection against data leaks, but they limit authentication opportunities (https for example, can't be used to connect embedded devices)

Author: Updated with latest merge

- Ignores other authentication needs such as between processes on one machine, or server-server relationships. 

	For example, setting up a payment processor like Stripe requires obtaining an API token and Key and connecting it to your product’s server. These keys can be compromised, creating security risks, and obtaining your key if it is lost is often impossible, which means starting the process from scratch.
- Storage of secret information is platform dependent
- API Key creation is also platform dependent for randomness

> - Offers limited language support based on each auth approach limiting usage to programs that use that language

Reviewer: ```suggestion
- Language support for a given auth approach might be limited, locking companies into a particular tech stack
```

- Requires users to be connected to the internet

**Solution**

Enter Sessionless, an authentication protocol (Sessionless Authentication Protocol, or SLAP), allowing for simplicity and flexibility by opening opportunities for indirect authentication. 

With Sessionless, Services (websites, applications, and other digital services) can:
- Authenticate users without gathering personal information which will
- Reduce bounce rates
- Remove security risks
- Eliminate need for platform dependency to store secret information
- Simplify authentication by removing key creation process 
- Authenticate users across devices such as computers, smart Tvs, Game Consoles, whether connected to the internet or connected via bluetooth devices

And Users can:
- Decrease usage of password managers
- Gain security 

> 

Reviewer: This part's great

> **Use Cases**

Reviewer: This feels partly finished, but I'm happy to add to it myself

Author: Oops, please do go ahead and add!
